src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -1355,24 +1356,35 @@ public Task DownloadUsersAsync(params IGuild[] guilds)
                                             return;
                                         }
 
-                                        SocketPresence before;
+                                        SocketPresence beforePresence;
+                                        SocketGuildUser beforeGuild;
+                                        SocketGlobalUser beforeGlobal;
                                         var user = guild.GetUser(data.User.Id);
                                         if (user != null)
                                         {
-                                            before = user.Presence.Clone();
+
+                                            beforePresence = user.Presence.Clone();
+                                            beforeGlobal = user.GlobalUser.Clone();
+                                            beforeGuild = user.Clone();
```

Comment:
MemberwiseClone is a shallow clone, which is the main reason we use it - deep clones would be expensive. Having to clone global is one of the side effects of that.

Memory-wise the operation is really cheap since it's just a memcpy iirc. It does involve memory pinning, but I've yet to see an impact.